backend/src/v2/metadata/client.go

```diff
@@ -846,14 +855,25 @@ func (c *Client) GetExecutionsInDAG(ctx context.Context, dag *DAG, pipeline *Pip
 		glog.V(4).Infof("execs: %v", execs)
 		for _, e := range execs {
 			execution := &Execution{execution: e}
-			taskName := execution.TaskName()
+			taskName := UpdateTaskNameWithDagID(execution.TaskName(), parentDAGID)
```

Comment:
It may be helpful to add a comment here explaining why this is necessary, e.g.:

```
// Sometimes components in nested DAGs have identical task names. We update the task name to include the DAG ID to avoid key collisions in the executions map.
```